### PR TITLE
Shallow submodules

### PR DESCRIPTION
**Pull request by zachgersh**
- Depth flag is respected during submodule clone
- Adds test coverage for depth flag in general

### CODE REVIEW

**Comment by zachgersh** [comment]
Don't merge yet.  Need to cover one more test case.

File: assets/in [hunk 1]
```diff
@@ -55,12 +55,14 @@ git checkout -q $ref
 git log -1 --oneline
 git clean --force --force -d
 
+echo submodules
```
Comment (vito) [inline assets/in]: leftover debugging?

**Comment by vito** [comment]
Guessing this is mergeable now? Or is that one test case still a WIP.

**Comment by zachgersh** [comment]
@vito - yeah, think this is totally good to go.

**Comment by xoebus** [comment]
Thanks!

**Pull request merged**